### ISSUE TITLE
is it necessary to add a description of each picture in the training set？ 

### ISSUE THREAD

**Issue by Jason-kid** [huggingface/diffusers#2622]
Hey, thanks for this great project, I'm a bit confused.
For painting style learning, is it necessary to add a description of each picture in the training set? For example, 1) use a unique word to represent the style of painting, such as <sks style> used in dreambooth lora, and there is no image description file; 2) use image-text pair data, such as in train_txt_image_lora, a picture corresponds to a description file.
What is the difference between them? Which method is more conducive to painting style learning and reasoning use?

**Comment by sayakpaul** [comment]
Hey there! Thanks for your kind words and your question. I believe it greatly serves the purpose of open-science :) 

> 1) use a unique word to represent the style of painting, such as used in dreambooth lora, and there is no image description file

Well, for DreamBooth the captions / description / prompts actually stay the same for all the images used for DreamBooth training. To make things concrete, here's a snapshot of the dataset we use in the DreamBooth [training example](https://github.com/huggingface/diffusers/tree/main/examples/dreambooth):

![image](https://user-images.githubusercontent.com/22957388/224609794-470d1685-c33a-40ea-ab10-b69c8c1fbbb4.png)

The `[instance_id]` (shown in the image above) we use in the example is "sks". As you can see, the prompts are one and the same and in the example, we refer to this prompt as `instance_prompt`:

https://github.com/huggingface/diffusers/blob/bbab8553224d12f7fd58b0e65b0daf899769ef0b/examples/dreambooth/train_dreambooth.py#L169

Along with this dataset, when [prior preservation](https://github.com/huggingface/diffusers/tree/main/examples/dreambooth#training-with-prior-preservation-loss) is enabled, we make use of the `class_prompt`:

https://github.com/huggingface/diffusers/blob/bbab8553224d12f7fd58b0e65b0daf899769ef0b/examples/dreambooth/train_dreambooth.py#L176

Even when using prior preservation, the same `class_prompt` is used for generating the "class images". 

In a sense, these datasets are still paired, i.e., an image is paired with some description / caption / prompt but those captions are one and the same. Whereas, in the vanilla [text-to-image case](https://github.com/huggingface/diffusers/blob/main/examples/text_to_image), the dataset has image-caption pairs where the captions are different for each image. Like [this](https://huggingface.co/datasets/lambdalabs/pokemon-blip-captions):

<img width="1021" alt="image" src="https://user-images.githubusercontent.com/22957388/224610789-b9546059-3729-44d9-9b21-7d872d50f484.png">

I hope the difference between the datasets is now clear.

**Comment by sayakpaul** [comment]
Coming to 

> What is the difference between them? Which method is more conducive to painting style learning and reasoning use?

I believe the primary difference lies in their objectives and scopes. DreamBooth is more targeted toward customizing the generation toward a very specific visual concept (like a specific type of face or any other object). 

But for the vanilla text-to-image fine-tuning, we are interested in being a bit more general toward the generation process. In [our example](https://github.com/huggingface/diffusers/blob/main/examples/text_to_image), we're interested to fine-tune the model such that the generated images appear in the visual context of Pokemons. We don't need to rely on using the instance identifier (like "sks") in our inference prompts here as opposed to DreamBooth. 

So, to answer your question, if you're aiming for generality, then I'd suggest going for regular text-to-image fine-tuning. We're working on InstructPix2Pix (https://github.com/huggingface/diffusers/pull/2478) that might interest you. If you want the generation to be ultra-specific, I'd suggest going for DreamBooth or even [Textual Inversion](https://github.com/huggingface/diffusers/blob/main/examples/textual_inversion/).

I hope it answers your question :-)

**Comment by Jason-kid** [comment]
> 

Thanks for your patience in answering, i get it and close the issue.

**Comment by edwardelric1202** [comment]
> Hey there! Thanks for your kind words and your question. I believe it greatly serves the purpose of open-science :)
> 
> > 1. use a unique word to represent the style of painting, such as used in dreambooth lora, and there is no image description file
> 
> Well, for DreamBooth the captions / description / prompts actually stay the same for all the images used for DreamBooth training. To make things concrete, here's a snapshot of the dataset we use in the DreamBooth [training example](https://github.com/huggingface/diffusers/tree/main/examples/dreambooth):
> 
> ![image](https://user-images.githubusercontent.com/22957388/224609794-470d1685-c33a-40ea-ab10-b69c8c1fbbb4.png)
> 
> The `[instance_id]` (shown in the image above) we use in the example is "sks". As you can see, the prompts are one and the same and in the example, we refer to this prompt as `instance_prompt`:
> 
> https://github.com/huggingface/diffusers/blob/bbab8553224d12f7fd58b0e65b0daf899769ef0b/examples/dreambooth/train_dreambooth.py#L169
> 
> Along with this dataset, when [prior preservation](https://github.com/huggingface/diffusers/tree/main/examples/dreambooth#training-with-prior-preservation-loss) is enabled, we make use of the `class_prompt`:
> 
> https://github.com/huggingface/diffusers/blob/bbab8553224d12f7fd58b0e65b0daf899769ef0b/examples/dreambooth/train_dreambooth.py#L176
> 
> Even when using prior preservation, the same `class_prompt` is used for generating the "class images".
> 
> In a sense, these datasets are still paired, i.e., an image is paired with some description / caption / prompt but those captions are one and the same. Whereas, in the vanilla [text-to-image case](https://github.com/huggingface/diffusers/blob/main/examples/text_to_image), the dataset has image-caption pairs where the captions are different for each image. Like [this](https://huggingface.co/datasets/lambdalabs/pokemon-blip-captions):
> 
> <img alt="image" width="1021" src="https://user-images.githubusercontent.com/22957388/224610789-b9546059-3729-44d9-9b21-7d872d50f484.png">
> 
> I hope the difference between the datasets is now clear.

Hi, I wonder if I can train multiple concepts in once dreambooth training. For example, I got some images of dogs and cats in pixelated style, and I want train a model that can generate like "a pixelated style cat" and "a pixelated dog", is it possible for diffusers now?

**Comment by sayakpaul** [comment]
No, we currently don't support DreamBoothing with multiple concepts.

**Comment by edwardelric1202** [comment]
> No, we currently don't support DreamBoothing with multiple concepts.

Ok, thanks for your reply.

**Comment by Adenialzz** [comment]
> Hey there! Thanks for your kind words and your question. I believe it greatly serves the purpose of open-science :)
> 
> > 1. use a unique word to represent the style of painting, such as used in dreambooth lora, and there is no image description file
> 
> Well, for DreamBooth the captions / description / prompts actually stay the same for all the images used for DreamBooth training. To make things concrete, here's a snapshot of the dataset we use in the DreamBooth [training example](https://github.com/huggingface/diffusers/tree/main/examples/dreambooth):
> 
> ![image](https://user-images.githubusercontent.com/22957388/224609794-470d1685-c33a-40ea-ab10-b69c8c1fbbb4.png)
> 
> The `[instance_id]` (shown in the image above) we use in the example is "sks". As you can see, the prompts are one and the same and in the example, we refer to this prompt as `instance_prompt`:
> 
> https://github.com/huggingface/diffusers/blob/bbab8553224d12f7fd58b0e65b0daf899769ef0b/examples/dreambooth/train_dreambooth.py#L169
> 
> Along with this dataset, when [prior preservation](https://github.com/huggingface/diffusers/tree/main/examples/dreambooth#training-with-prior-preservation-loss) is enabled, we make use of the `class_prompt`:
> 
> https://github.com/huggingface/diffusers/blob/bbab8553224d12f7fd58b0e65b0daf899769ef0b/examples/dreambooth/train_dreambooth.py#L176
> 
> Even when using prior preservation, the same `class_prompt` is used for generating the "class images".
> 
> In a sense, these datasets are still paired, i.e., an image is paired with some description / caption / prompt but those captions are one and the same. Whereas, in the vanilla [text-to-image case](https://github.com/huggingface/diffusers/blob/main/examples/text_to_image), the dataset has image-caption pairs where the captions are different for each image. Like [this](https://huggingface.co/datasets/lambdalabs/pokemon-blip-captions):
> 
> <img alt="image" width="1021" src="https://user-images.githubusercontent.com/22957388/224610789-b9546059-3729-44d9-9b21-7d872d50f484.png">
> 
> I hope the difference between the datasets is now clear.

Hi, thanks for your explanation. But when we train dreambooth about specific style, we may want to attach semantic captions describing the image content along with the style keyword, like: `a girl standing on the street, sks style`. So how can we adapt DreamBoothDataset to text-image-pair format ?
